app/js/disease-associations-controller.js

```diff
@@ -150,12 +164,123 @@ angular.module('cttvControllers')
     // on PAGE LOAD
     //
 
+    cttvUtils.clearErrors();
+    $scope.filters = cttvLocationState.getState()[facetsId] || {};
+    render(cttvLocationState.getState(), cttvLocationState.getOldState());
 
+    $scope.uploadedFile = function (element) {
+        $scope.$apply(function ($scope) {
+            $scope.files = element.files;
+        });
 
-    cttvUtils.clearErrors();
-    $scope.filters = cttvLocationState.getState()[facetsId] || {} ;
-    render( cttvLocationState.getState(), cttvLocationState.getOldState() );
+        $scope.addFile();
+    };
+
+    $scope.removeTargets = function(){
+        var theElement = document.getElementById("myFileInput");
+
+        theElement.value = null;
+        initFilterByFile();
+        //console.log("removeTargets:$scope.files=", $scope.files );
+        getFacets($scope.filters);
+    };
+
+    $scope.addFile = function () {
+        $scope.validateFile($scope.files[0]);
+    };
+
+    $scope.validateFile = function (file) {
+        $scope.fileName = file.name;
+        var reader = new FileReader();
+        reader.onloadend = function (evt) {
+            //do something with file content here
+            var myFileContent = evt.target.result;
+            $scope.targetNameArray = myFileContent.replace(/(\r\n|\n|\r)/gm, '\n').split('\n');
+            $scope.targetNameArray = $scope.targetNameArray.filter(function(e){ return e.trim();}); //get rid of empty strings
+
+            getTargetIds();
+        };
+        reader.readAsText(file);
+    };
+
+    var getTargetIds = function() {
+        var promise = $q(function (resolve) {
+            resolve("");
+        });
+
+        $scope.targetNameArray.forEach(function (targetName){
+            promise = promise.then(function() {
+               return getTargetId(targetName);
```

Comment:
This makes the searches synchronously!
Why don't you create an array of promises and wait for all of them to complete?
something like...
var searchPromiseArray = [];
[...]
searchPromiseArray.push(getTargetId(targetName);
[...]
$q.all(searchPromiseArray).then(...);